common/web/keyboard-processor/src/keyboards/activeLayout.ts

```diff
@@ -346,22 +353,25 @@ namespace com.keyman.keyboards {
       // Allow for right OSK margin (15 layout units)
       let rightMargin = ActiveKey.DEFAULT_RIGHT_MARGIN/totalWidth;
 
-      // If a single key, and padding is negative, add padding to right align the key
-      if(keys.length == 1 && parseInt(keys[0]['pad'],10) < 0) {
-        keyPercent=parseInt(keys[0]['width'],10)/totalWidth;
-        keys[0]['widthpc']=keyPercent;
-        keys[0]['padpc']=1-(totalPercent + keyPercent + rightMargin);
+      if(keys.length > 0) {
+        const finalKey = keys[keys.length-1] as ActiveKey;
 
-        // compute center's default x-coord (used in headless modes)
-        setProportions(keys[0] as ActiveKey, padPercent, keyPercent, totalPercent);
-      } else if(keys.length > 0) {
-        let j=keys.length-1;
-        padPercent=parseInt(keys[j]['pad'],10)/totalWidth;
-        keys[j]['padpc']=padPercent;
-        keys[j]['widthpc'] = keyPercent = 1-(totalPercent + padPercent + rightMargin);
+        // If a single key, and padding is negative, add padding to right align the key
+        if(keys.length == 1 && finalKey.pad < 0) {
+          keyPercent = finalKey.width/totalWidth;
+          finalKey['widthpc']=keyPercent;
+          finalKey['padpc']=1-(totalPercent + keyPercent + rightMargin);
 
-        // compute center's default x-coord (used in headless modes)
-        setProportions(keys[j] as ActiveKey, padPercent, keyPercent, totalPercent);
+          // compute center's default x-coord (used in headless modes)
+          setProportions(finalKey, padPercent, keyPercent, totalPercent);
```

Comment:
`padPercent` here is no longer valid afaict? Or is the value dropping out of the loop correct? If so, that's a little too buried for my sanity!